bridge/scripts/tasks/alicenetTasks.ts

```diff
@@ -1750,6 +1751,251 @@ task("update-alicenet-node-version", "Set the Canonical AliceNet Node Version")
+      factory.signer.getAddress()
+    );
+    const gas = await factory.estimateGas.multiCall(
+      encodedMultiCallArgs,
+      await getGasPrices(hre.ethers)
+    );
+    promptMessage = `total gas required is ${gas.toString()} Continue? (y/n)\n`;
+    await promptCheckDeploymentArgs(promptMessage);
+    const txResponse = await factory.multiCall(
+      encodedMultiCallArgs,
+      await getGasPrices(hre.ethers)
+    );
+    const receipt = await txResponse.wait(waitConfirmationsBlocks);
+    const balanceAfter = await hre.ethers.provider.getBalance(
+      factory.signer.getAddress()
+    );
+    if (taskArgs.test) {
+      console.log(`total cost = ${balanceBefore.sub(balanceAfter)}`);
+    }
+    console.log(receipt.events);
+  });
+
+task(
+  "monthly-distribution",
+  "takes in a csv file with addresses, and ALCA amount"
+)
+  .addParam("csvPath", "path to csv file")
+  .addFlag("test", "test mode for use with a hardhat fork mode")
+  .addOptionalParam(
+    "waitConfirmation",
+    "wait specified number of blocks between transactions",
+    0,
+    types.int
+  )
+  .setAction(async (taskArgs, hre) => {
+    const waitConfirmationsBlocks = await parseWaitConfirmationInterval(
+      taskArgs.waitConfirmation,
+      hre
+    );
+    const maxStakingLock = BigNumber.from("1051200");
+    let factory = await hre.ethers.getContractAt(
+      ALICENET_FACTORY,
+      "0x4b6dF6B299fB6414f45719E0d9e1889269a7843E"
+    );
+    const publicStaking = await hre.ethers.getContractAt(
+      "PublicStaking",
+      "0x65683990415A669a7ecbD877240818EE458d0f09"
+    );
+    const alca = await hre.ethers.getContractAt(
+      "ALCA",
+      "0xBb556b0eE2CBd89ed95DdEA881477723A3Aa8F8b"
+    );
+    const distributionData = await readCSV(taskArgs.csvPath);
+    // use this flag with a hardhat forked node
+    if (taskArgs.test) {
+      const address = "0xff55549a3ceea32fba4794bf1a649a2363fcda53";
+      await hre.network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [address],
+      });
+      const signers = await hre.ethers.getSigners();
+      const tx = await signers[0].populateTransaction({
+        to: address,
+        value: hre.ethers.utils.parseEther("2"),
+      });
+      await signers[0].sendTransaction(tx);
+      const helpers = require("@nomicfoundation/hardhat-network-helpers");
+      await helpers.impersonateAccount(address);
+      const signer = await hre.ethers.getSigner(address);
+      factory = factory.connect(signer);
+    }
+    const encodedMultiCallArgs: Array<MultiCallArgsStruct> = [];
+    let encodedApprovalCall: MultiCallArgsStruct = {
+      target: alca.address,
+      value: 0,
+      data: "",
+    };
+    encodedMultiCallArgs.push(encodedApprovalCall);
+    let approvalAmount = BigNumber.from("0");
+    // encode a multicall input for each row in the csv file
+    for (const input of distributionData) {
+      if (input.Address.length !== 42) {
```

Comment:
Maybe you can use: ethers.utils.parseAddress() (something like that, I don't fully recall the name) here to check if you are getting a valid address.